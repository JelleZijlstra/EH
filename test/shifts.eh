#!/usr/bin/ehi
# The shift operators

class Foo
	operator<< = func: x -> (echo 'Left shift' + x)
	operator>> = func: x -> (echo 'Right shift' + x)
end

f = Foo.new()
f << 3
f >> 3
f <<= 3
printvar f
