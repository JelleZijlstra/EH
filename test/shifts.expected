Left shift3
Right shift3
Left shift3
null
